### PR TITLE
[KV] Fix kv:namespace create name validation

### LINKED ISSUE #4248: `wrangler kv:namespace create <namespace>` treats `<namespace>` as a binding name rather than the resource name
### Describe the solution

Currently, when creating a new KV namespace via Wrangler, it tries to be helpful and print out a snippet for the user to paste into their `wrangler.toml` file. 
```shell
wrangler kv:namespace create "MY_KV"
🌀 Creating namespace with title "worker-MY_KV"
✨ Success!
Add the following to your configuration file in your kv_namespaces array:
kv_namespaces = [
  { binding = "MY_KV", id = "e29b263ab50e42ce9b637fa8370175e8" }
]
```

But notice that we're treating what the user expects to be their _resource name_ as the name of the binding, with the actual name given to the resource being some decorated variation of that. As a result, we also artificially limit what a user can name their resource via this command by what constitutes a valid JS identifier (e.g. no dashes are allowed here, though KV namespaces can indeed contain dashes in their name).

We should fix this so that the resource is created with the name the user intended. We should output the resource name along with its ID. We can still output a helpful snippet for the user to paste into `wrangler.toml`, albeit leaving it up to them to replace the binding name with what they want.

```shell
wrangler kv:namespace create "my-kv-namespace"
🌀 Creating KV namespace...
✨ Success!
Created the following KV namespace:
    my-kv-namespace: e29b263ab50e42ce9b637fa8370175e8

To start using this resource in your worker, add an entry to `kv_namespaces` in your wrangler.toml file:

kv_namespaces = [
  { binding = "<VARIABLE_NAME>" id = "e29b263ab50e42ce9b637fa8370175e8" }
]
```

### PR DESCRIPTION
**Pull request by nora-soderlund**
Fixes #4248

**What this PR solves / how to test:** I came across this today which felt very odd. I couldn't name my namespace `myproduct-prod` because `wrangler kv:namespace create <namespace>` was expecting `<namespace>` to be a binding name.

image

Yet it created the namespace with a `worker-` prefix when I changed my input to `myproduct_prod`. 

image

Today, it doesn't do anything with the binding name, although in the code it's indicated that it will automatically add itself to wrangler.toml in the future. I believe the better approach for differencing between the namespace name and binding name is to opt for another input for the binding name (or as suggested in the code, create a stripped version of the input).

In the meantime, here's a PR which removes the _binding_ name check and replaces it instead with nothing, since it seems as if anything is acceptable as a namespace name in the dashboard.

image

**Author has addressed the following:** This PR is a breakout from #5007 to leave that PR open for an unrelated breaking change.

- Tests
  - [X] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [X] Included
  - [ ] Not necessary because:
- Associated docs
  - [ ] Issue(s)/PR(s): 
  - [x] Not necessary because: the docs do not mention the overly zealous validation of the title passed to the command.

### CODE REVIEW

File: .changeset/lovely-dolls-taste.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": minor
```
Comment (petebacondarwin) [inline .changeset/lovely-dolls-taste.md:2]: Not really a new feature, it's a bug fix, so I think we can go with a patch bump here.

File: .changeset/lovely-dolls-taste.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+fix: kv:namespace create to refer to the namespace name instead of the binding name
```
Comment (petebacondarwin) [inline .changeset/lovely-dolls-taste.md:5]: Let's expand this description a bit. Without any context I think the current text is a bit cryptic. Perhaps something like:

```
fix: allow `kv:namespace create` to accept a namespace name that contains characters not allowed in a binding name

This command tries to use the namespace name as the binding. Previously, we would unnecessarily error if this namespace name did not fit the binding name constraints. Now we accept such names and then remove invalid characters when generating the binding name.
```

**Review by petebacondarwin — APPROVED** [review]
Magic! Thanks for adding the extra unit tests for the new function.
Can we consider updating the changeset before landing?
Otherwise LGTM!

File: .changeset/lovely-dolls-taste.md (hunk 2 above)
Comment (nora-soderlund) [reply .changeset/lovely-dolls-taste.md]: For sure. Thanks. :)

**Pull request merged**